File: taskmanager-frontend/src/types/team.ts
export interface Team {
  id: number;
  name: string;
  description?: string;
  createdAt: string;
  members: Array<{
    id: number;
    name: string;
    email: string;
  }>;
}

export interface CreateTeamRequest {
  name: string;
  description?: string;
}

export type UpdateTeamRequest = CreateTeamRequest; 

File: taskmanager/src/main/java/com/tasksphere/taskmanager/domain/entity/Team.java
package com.tasksphere.taskmanager.domain.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.util.*;

@Entity
@Table(name = "teams")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Team {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String name;
    
    private String description;
    
    @OneToMany(mappedBy = "team")
    @Builder.Default
    private Set<User> members = new HashSet<>();
    
    @ManyToOne
    @JoinColumn(name = "team_lead_id")
    private User teamLead;
    
    @Column(nullable = false)
    private LocalDateTime createdAt;
    
    private LocalDateTime updatedAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
} 